src/components/labels.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Typography } from '@material-ui/core';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+
+interface LabelProps {
+  hideText: string;
+  modalLabelClass: string;
+  modeText?: string;
+  liftState?: Readonly<RomiCore.LiftState>;
+  liftName?: string;
+  name?: string;
+}
+
+export default function Labels(props: LabelProps) {
+
+  const { hideText, modalLabelClass, modeText, liftState, name, liftName } = props;
+
+  return (
+    <React.Fragment>
+      <Typography variant="h6" className={hideText}>
+        {liftName ? liftName : name}
+      </Typography>
+      {
+        liftName ? 
+        <Typography className={modalLabelClass} variant="button">
+          {liftState ? liftState.current_floor : 'N/A'}
+        </Typography>
+        :
+        <Typography className={modalLabelClass} variant='button'>
+          {modeText}
+      </Typography>
+      }
+    </React.Fragment>
+  )
+}
```

Comment:
Components like these ([presentational](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)) should ideally be independent of the logic of other components. In other words, the component should be responsible for rendering a label with plain data passed to it and not knowing if there is an elevator status. So I would recommend that the elevator status be handled in the parent component. I mean sending 
`{liftState? liftState.current_floor: 'N / A'} ` in the `modeText` parameter.

Reply:
Yea, the label shouldn't need information on the lift/door/robot state, it should just receive a string for the text. I don't think we need 3 `Typography` components that render conditionally either.